InteractiveHtmlBom/web/render.js

```diff
@@ -632,6 +640,43 @@ function pointWithinDistanceToSegment(x, y, x1, y1, x2, y2, d) {
   return dx * dx + dy * dy <= d * d;
 }
 
+function modulo(n, mod) {
+  return ((n % mod) + mod ) % mod;
+}
+
+function pointWithinDistanceToArc(x, y, xc, yc, radius, startangle, endangle, d) {
+  var dx = x - xc;
+  var dy = y - yc;
+  var r_sq = dx * dx + dy * dy;
+  var rmin = Math.max(0,radius-d);
+  var rmax = (radius+d);
```

Comment:
Please format like the rest of expressions. Spaces around operands and after comma, no unnecessary brackets.

Reply:
Thanks for your patience.  I'm sure it's obvious that I'm not a programmer by trade...